### PR TITLE
feat(HMS-2439): Add private IP address in Azure details

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Fetch Azure private IP address as well as public one.

### CODE REVIEW

**Review by lzap — CHANGES_REQUESTED** [review]
Something is not right, this should not pass since you haven’t modified OpenAPI. What is wrong? This should fail.

**Comment by lzap** [comment]
Aaah I understand now, the PrivateIPv4 is a shared model, no need to change OpenAPI.

The only concern I have is the nil exceptions which can possibly occur. It is very unlikely that an internal IPv4 is not set, not in 2023, it is still another two decades until full IPv6 adoption.

**Comment by lzap** [comment]
Can you add example similarly as I did in AWS please?

**Comment by ezr-ondrej** [comment]
Added example and check for nil pointer :)

**Comment by ezr-ondrej** [comment]
> Aaah I understand now, the PrivateIPv4 is a shared model, no need to change OpenAPI.

Yup, that's why I was so keen to implement this, the hard part was already done by you! xD

File: internal/clients/http/azure/create_vms.go [hunk 1]
```diff
@@ -38,6 +38,11 @@ func (c *client) CreateVMs(ctx context.Context, vmParams clients.AzureInstancePa
 		}
 
 		vmDescriptions[i].IPv4 = *publicIP.Properties.IPAddress
+		// can be nil if IPv6 is default in the future
+		privateIPv4 := networkInterface.Properties.IPConfigurations[0].Properties.PrivateIPAddress
```
Comment (avitova) [inline internal/clients/http/azure/create_vms.go:42]: > 		// can be nil if IPv6 is default in the future

If this happends, would it end up with only PrivateIPAddress being nil? Everything else here are pointers/accessing IPConfigartions array that might be empty. Is this all safe?:D

Comment (lzap) [reply internal/clients/http/azure/create_vms.go:42]: We have a `ptr.FromOrEmpty` generic helper, can you use that instead?

Comment (ezr-ondrej) [reply internal/clients/http/azure/create_vms.go:42]: it is not safe for sure xD it's YOLO I'll try to improve at least the ones I'm touching.
But the structure should be pretty stable, Azure uses pointers everywhere, but it either fails, or the data are there, so it's "a bit safer then complete YOLO" xD

Comment (ezr-ondrej) [reply internal/clients/http/azure/create_vms.go:42]: Will use the `ptr.FromOrEmpty` :+1:

Comment (ezr-ondrej) [reply internal/clients/http/azure/create_vms.go]: Done, I'll try to fix-up other occurences when I touch other Azure code :)

**Review by lzap — APPROVED** [review]
Thanks, merge at will @avitova

File: internal/clients/http/azure/create_vms.go (hunk 1 above)
Comment (avitova) [reply internal/clients/http/azure/create_vms.go]: Yeah, I do believe that for Azure this would not be inconsistent.

**Comment by avitova** [comment]
Thanks! LGTM:)

**Pull request merged**